granulate_utils/linux/cgroups/cgroup.py

```diff
@@ -88,13 +198,46 @@ def get_cgroup_mount(controller: str, resolve_host_root_links: bool = True) -> O
     v2_path = find_v2_hierarchy(resolve_host_root_links)
     if v1_paths and controller in v1_paths:
         # Either v1 or hybrid with the requested controller bound to a v1 hierarchy
-        return v1_paths[controller]
+        return CgroupCoreV1(Path(v1_paths[controller]))
     if v2_path:
-        # v2 (unified) - all controllers are in a single unified hierarchy
-        return v2_path
+        # v2 (unified) - check given controller is supported in cgroup v2
+        cgroup_v2 = CgroupCoreV2(Path(v2_path))
+        if cgroup_v2.is_controller_supported(controller):
+            return cgroup_v2
     # No cgroup mount of the requested controller
     return None
 
 
-def is_known_controller(controller: str) -> bool:
-    return controller in SUBSYSTEMS
+def get_cgroup_mount_checked(controller: str, resolve_host_root_links: bool = True) -> CgroupCore:
+    cgroup_mount = get_cgroup_mount(controller)
+    assert cgroup_mount is not None, f"Could not find cgroup mount point for controller {controller!r}"
```

Comment:
This checks what I commented [here](https://github.com/Granulate/granulate-utils/pull/117/files#r1089819591) I suppose. I'd say it's not an `assert` - asserts are for things which are bugs in your code, while this case ain't a bug - if our process runs in a cgroup where the parent cgroup already controls the controller we wish... we should be notified of it & fix the problem, possibly automatically in the code.

OR I misunderstood this concept, will comment here back if I understand it better, or you let me know if I'm wrong

Reply:
The case you mentioned here shouldn't matter as we're looking at the root cgroup, our current process doesn't matter here.
Maybe it shouldn't be an assert and it should raise a `ControllerNotMounted` exception instead.